msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```diff
@@ -221,15 +302,177 @@ public static IMultipleAccountPublicClientApplication createMultipleAccountPubli
+                                                                    @NonNull final ApplicationCreatedListener listener) {
+
+        validateNonNullArgument(context, "context");
+        validateNonNullArgument(listener, "listener");
         if (context == null) {
             throw new IllegalArgumentException("context is null.");
         }
 
-        return new MultipleAccountPublicClientApplication(context, loadConfiguration(configFile));
+        create(context,
+                loadConfiguration(context, configFileResourceId),
+                listener
+        );
     }
 
+    /**
+     * {@link PublicClientApplication#createSingleAccountPublicClientApplication(Context, int)} will read the client id and other configuration settings from the
+     * file included in your applications resources.
+     * <p>
+     * For more information on adding configuration files to your applications resources please
+     *
+     * @param context              Application's {@link Context}. The sdk requires the application context to be passed in
+     *                             {@link PublicClientApplication}. Cannot be null.
+     *                             <p>
+     *                             Note: The {@link Context} should be the application context instead of the running activity's context, which could potentially make the sdk hold a
+     *                             strong reference to the activity, thus preventing correct garbage collection and causing bugs.
+     *                             </p>
+     * @param configFileResourceId The resource ID of the raw file containing the JSON configuration for the PublicClientApplication
+     * @return An instance of ISingleAccountPublicClientApplication
+     * @see <a href="https://developer.android.com/guide/topics/resources/providing-resources">Android app resource overview</a>
+     * @see <a href="https://github.com/AzureAD/microsoft-authentication-library-for-android/wiki">MSAL Github Wiki</a>
+     */
+    @WorkerThread
+    public static ISingleAccountPublicClientApplication createSingleAccountPublicClientApplication(@NonNull final Context context,
+                                                                                                       final int configFileResourceId) throws InterruptedException, MsalException {
+        validateNonNullArgument(context, "context");
+
+        final PublicClientApplicationConfiguration configuration = loadConfiguration(context, configFileResourceId);
+
+        IPublicClientApplication application = createPublicClientApplication(context, configuration);
+        if(application instanceof ISingleAccountPublicClientApplication){
+            return (ISingleAccountPublicClientApplication) application;
+        }else{
+            if(configuration.mAccountMode != AccountMode.SINGLE){
+                throw new MsalClientException("AccountMode in configuration is not set to single");
```

Comment:
nit: move this above createPublicClientApplication()

Reply:
Fixed.